client/src/test/java/com/sinch/sdk/e2e/domains/conversation/AppsSteps.java

```diff
@@ -0,0 +1,140 @@
+package com.sinch.sdk.e2e.domains.conversation;
+
+import com.sinch.sdk.domains.conversation.api.v1.AppService;
+import com.sinch.sdk.domains.conversation.models.v1.ProcessingMode;
+import com.sinch.sdk.domains.conversation.models.v1.app.ConversationMetadataReportView;
+import com.sinch.sdk.domains.conversation.models.v1.app.request.AppCreateRequest;
+import com.sinch.sdk.domains.conversation.models.v1.app.request.AppUpdateRequest;
+import com.sinch.sdk.domains.conversation.models.v1.app.request.ConversationChannelCredentialRequest;
+import com.sinch.sdk.domains.conversation.models.v1.app.request.ConversationChannelCredentialsRequestBuilderFactory;
+import com.sinch.sdk.domains.conversation.models.v1.app.response.AppResponse;
+import com.sinch.sdk.domains.conversation.models.v1.credentials.ChannelIntegrationStatus;
+import com.sinch.sdk.domains.conversation.models.v1.credentials.StaticBearerCredentials;
+import com.sinch.sdk.e2e.Config;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import java.util.Arrays;
+import java.util.Collection;
+import org.junit.jupiter.api.Assertions;
+
+public class AppsSteps {
+
+  AppService service;
+  AppResponse createResponse;
+  Collection<AppResponse> listResponse;
+  AppResponse getResponse;
+  AppResponse updateResponse;
+  boolean deletePassed;
+
+  @Given("^the Conversation service \"Apps\" is available$")
+  public void serviceAvailable() {
+    service = Config.getSinchClient().conversation().v1().app();
+  }
+
+  @When("^I send a request to create an app$")
+  public void create() {
+
+    String identity = "SpaceMonkeySquadron";
+    String token = "00112233445566778899aabbccddeeff";
+    String displayName = "E2E Conversation App";
+
+    ConversationChannelCredentialRequest credentials =
+        ConversationChannelCredentialsRequestBuilderFactory.sms(
+                StaticBearerCredentials.builder()
+                    .setClaimedIdentity(identity)
+                    .setToken(token)
+                    .build())
+            .build();
+
+    AppCreateRequest request =
+        AppCreateRequest.builder()
+            .setChannelCredentials(Arrays.asList(credentials))
+            .setDisplayName(displayName)
+            .build();
+
+    createResponse = service.create(request);
+  }
+
+  @When("^I send a request to list all the apps$")
+  public void list() {
+
+    listResponse = service.list();
+  }
+
+  @When("^I send a request to retrieve an app$")
+  public void get() {
+
+    getResponse = service.get("foo");
+  }
+
+  @When("^I send a request to update an app$")
+  public void update() {
+
+    AppUpdateRequest request =
+        AppUpdateRequest.builder().setDisplayName(" a new display name").build();
+    updateResponse = service.update("foo", request);
+  }
+
+  @When("^I send a request to delete an app$")
+  public void delete() {
+
+    service.delete("foo");
+    deletePassed = true;
```

Comment:
Here you are just checking that the request completed without exception right?
Note that if we decide to move to static values instead of regexp for parameter matching, this test will fail as `foo` is certainly not a value corresponding to an appId. You may reuse the one tou are already checking in the common fields

Reply:
According to our discussion: we agree to update tests accordingly in case of future change